sync/etcdv3/etcd_test.go

```diff
@@ -277,88 +233,140 @@ func TestWatchWithRevision(t *testing.T) {
 		t.Fatalf("mismatch response: %+v, expecting /new, existing==false, revision==%d", resp, secondRevision)
 	}
 
-	putResponse, err = sync.etcdClient.Put(ctx, path+"/third", `{"existing": false}`)
-	if err != nil {
-		t.Fatalf("failed to update key: %s", err)
-	}
-	thirdRevision := putResponse.Header.Revision
+	thirdRevision := sync.mustPut(path+"/third", `{"existing": false}`)
 
 	resp = <-responseChan
 	if resp.Key != path+"/third" || resp.Data["existing"].(bool) != false || resp.Revision != thirdRevision {
 		t.Fatalf("mismatch response: %+v, expecting /third, existing==false, revision==%d", resp, thirdRevision)
 	}
-
 }
 
-func TestFetchMultipleNodes(t *testing.T) {
-	ctx := context.Background()
+func TestShouldReturnAllValuesWhenWatchingAtCurrentRevision(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
consider context.WithTimeout(), if sync would hang test will be interrupted after 10 minutes.

Reply:
good point, changed